employees/views.py

```diff
@@ -48,7 +52,18 @@ def perform_create(self, serializer: ReportSerializer) -> None:
         serializer.save(author=self.request.user)
 
 
-class ReportList(APIView):
+@method_decorator(login_required, name="dispatch")
+@method_decorator(
+    check_permissions(
+        allowed_user_types=[
+            CustomUser.UserType.EMPLOYEE.name,
+            CustomUser.UserType.MANAGER.name,
+            CustomUser.UserType.ADMIN.name,
+        ]
+    ),
+    name="dispatch",
+)
+class ReportList(APIView, UserIsManagerOfCurrentReportProjectMixin, UserIsAuthorOfCurrentReportMixin):
```

Comment:
My bad and good catch, should be like in `ReportDetail`, but still it won't work with `APIView` which doesn't use `get_queryset` method, but it is in progress of refactoring to django view by @Szymiks so it can wait for his pull request (or got merged, as it doesn't break anything).

Reply:
sounds fine to me ;)